library/src/main/java/com/github/pwittchen/prefser/library/Prefser.java

```diff
@@ -253,20 +247,20 @@ public boolean contains(String key) {
     Preconditions.checkNotNull(key, KEY_IS_NULL);
     Preconditions.checkNotNull(typeTokenOfT, TYPE_TOKEN_OF_T_IS_NULL);
 
+    if (!contains(key)) {
+      return defaultValue;
```

Comment:
@pwittchen here we are returning the defValue, so if the execution reaches there (`return preferences.getBoolean(key, (defaultValue == null) ? false : defaultValue);`) SharedPreferences will contain the key and defValue will not be used.